lib/spack/spack/package.py

```diff
@@ -782,11 +790,53 @@ def stage(self, stage):
 
     @property
     def env_path(self):
-        return os.path.join(self.stage.path, 'spack-build.env')
+        """The Spack build environment file path."""
+        # Backward compatibility: Return the name of an existing log path;
+        # otherwise, return the current install env path name.
+        old_filename = os.path.join(self.stage.path, 'spack-build.env')
+        if os.path.exists(old_filename):
+            return old_filename
+        else:
+            return os.path.join(self.stage.path, _spack_build_envfile)
+
+    @property
+    def install_env_path(self):
+        """Return the path to the install env file."""
+        install_path = spack.store.layout.metadata_path(self.spec)
+
+        # Backward compatibility: Return the name of an existing log path;
+        # otherwise, return the current install env path name.
+        old_filename = os.path.join(install_path, 'build.env')
+        if os.path.exists(old_filename):
+            return old_filename
+        else:
+            return os.path.join(install_path, _spack_build_envfile)
 
     @property
     def log_path(self):
-        return os.path.join(self.stage.path, 'spack-build.txt')
+        """The Spack build log file path."""
```

Comment:
Could you update the docstrings of `log_path`, `install_log_path`, and `build_log_path` to explain how they differ and give a sense of when they are used?